apm/apm.go

```diff
@@ -16,14 +16,36 @@ import (
 )
 
 type Apm struct {
-	Logger logger.Logger
+	Logger *logger.Logger
 }
 
-func NewApm() Apm {
-	logger := logger.NewLogger()
-	return Apm{
-		Logger: logger,
+type ApmOption func(*Apm)
+
+func WithLogger(logger *logger.Logger) ApmOption {
```

Comment:
```suggestion
func WithLogger(logger logger.Logger) ApmOption {
```
make a copy  of the logger, to ignore changes from the outside on the injected logger